ygen/codegen.go

```diff
@@ -796,29 +800,27 @@ func mappableLeaf(e *yang.Entry) *yang.Entry {
 	return nil
 }
 
-// findMappableEntties finds the descendants of a yang.Entry (e) that should be mapped in
+// findMappableEntities finds the descendants of a yang.Entry (e) that should be mapped in
 // the generated code. The descendants that represent directories are appended to the dirs
 // map (keyed by the schema path). Those that represent enumerated types (identityref, enumeration,
 // unions containing these types, or typedefs containing these types) are appended to the
 // enums map, which is again keyed by schema path. If any child of the entry is in a module
 // defined in excludeModules, it is skipped. If compressPaths is set to true, then names are
-// mapped with path compression enabled.
-func findMappableEntities(e *yang.Entry, dirs map[string]*yang.Entry, enums map[string]*yang.Entry, excludeModules []string, compressPaths bool) {
-	pp := strings.Split(e.Path(), "/")
-	if !strings.HasPrefix(e.Path(), "/") {
-		// This is a malformed path, since the path is defined in the form
-		// /module/container-one.
-		return
-	}
-
+// mapped with path compression enabled. The set of modules that the current code generation
+// is processing is specified by the modules slice. This function returns slice of errors
+// encountered during processing.
+func findMappableEntities(e *yang.Entry, dirs map[string]*yang.Entry, enums map[string]*yang.Entry, excludeModules []string, compressPaths bool, modules []*yang.Entry) util.Errors {
 	// Skip entities who are defined within a module that we have been instructed
 	// not to generate code for.
 	for _, s := range excludeModules {
-		if s == pp[1] {
-			return
+		for _, m := range modules {
+			if m.Name == s && m.Namespace().Name == e.Namespace().Name {
+				return nil
+			}
 		}
 	}
 
+	var errs []error
```

Comment:
Since util.Errors is being returned, it's better to use that type and associated functions, which take care of those annoying nil errors appearing in error lists among other things.

Reply:
Done -- and throughout this file.